azurelinuxagent/common/version.py

```diff
@@ -116,6 +117,24 @@ def get_distro():
     return osinfo
 
 
+def get_lis_version():
+    """
+    This uses the Linux kernel's 'modinfo' command to retrieve the "version"
+    field for the "hv_vmbus" kernel module (the LIS drivers). This is the
+    documented method to retrieve the LIS version.
+    """
+    try:
+        lis_version = shellutil.run_command(["modinfo", "-F", "version", "hv_vmbus"], log_error=True)
+        # If the system doesn't have LIS drivers, 'modinfo' will
+        # return nothing on stdout, which will cause 'run_command' to
+        # return an empty string.
+        if lis_version:
+            return lis_version
+        return "Unknown"
+    except: # pylint: disable=W0702
```

Comment:
NIT: A missed linter comment here

Reply:
Also, pure synthetic comment, but you could have a single `return "Unknown"` after the `except` call rather than 2 returns for the same value. Functionally its the same just looks a bit more cleaner. (I'm good with the code either way)

Eg: 

```python
try:
     do_something()
     return_val = "val"
except: 
     # Error handling
     return_val = "error"
return return_val
 ```

```

Reply:
It's intentionally catching and ignoring all exceptions here (which results in a pylint warning) since this code runs really early at startup (I followed the pattern in `get_distro()` above which implied these code paths want _zero_ exceptions being raised).

Reply:
```python
        try:
            # dist() removed in Python 3.8
            osinfo = list(platform.dist()) + [''] # pylint: disable=W1505,E1101
        except: # pylint: disable=W0702
            osinfo = ['UNKNOWN', 'FFFF', '', '']
```

Reply:
Ahh I see. Could you still just have a blanket catch and just log it as a warning or something? It would be useful info incase we ever need to debug. That would also take care of the pylint warning

Reply:
I'm not a fan of that style because it relies on Python's weird concept of scope. For full clarity it would need to be:

```python
    lis_version = "Unknown"
    try:
        command_output = shellutil.run_command(["modinfo", "-F", "version", "hv_vmbus"], log_error=True)
        # If the system doesn't have LIS drivers, 'modinfo' will
        # return nothing on stdout, which will cause 'run_command' to
        # return an empty string.
        if command_output:
            lis_version = command_output
    except: # pylint: disable=W0702
        # Ignore every possible exception because this is in a
        # critical code path.
        pass
    return lis_version
```

Do you prefer that?

Reply:
As @narrieta mentioned in another comment, this is all legacy code that we're in the process of rewriting bit by bit, so dont worry about maintaining consistency with this code :)

Reply:
It might be reasonable in both this function and `get-distro()` to catch `Exception`, which in the [hierarchy](https://docs.python.org/3.5/library/exceptions.html#exception-hierarchy) is the base exception for everything but SystemExit, KeyboardInterrupt, and GeneratorExit.

Reply:
Yeah I see that, lmao, I was doing my best to fit my code into the existing as closely as possible, rather than creating new patterns and precedents in code I don't maintain 😅

Reply:
> Do you prefer that?

This looks good to me! 

Also catching `Exception` (not `BaseException`) sounds good 👍

Reply:
You got it! 😀

Reply:
> Yeah I see that, lmao, I was doing my best to fit my code into the existing as closely as possible, rather than creating new patterns and precedents in code I don't maintain 😅

haha no worries, thanks for being considerate towards our code :)